### DISCUSSION TITLE
No such mapping error when source the init file. When have the autocommand to source the config file

### DISCUSSION THREAD

**Discussion by OriLiMu** [LazyVim/LazyVim#2148] [Q&A] [answered]
I use the wildfire plugin in nvim. 
This plugin has a keymap for <CR> which I don't want. 
Related code you can check [here](https://github.com/gcmt/wildfire.vim/blob/master/plugin/wildfire.vim#L37). 
So I want to unmap it When I use the code 
```lua
local unmap = vim.keymap.del
unmap("n", "<CR>")
```
It works fine in normal case. 
But I have an auto command for source the config when save it. The code is: 
```lua
vim.api.nvim_create_augroup("source config", {})
vim.api.nvim_clear_autocmds({ group = "source config" })
vim.api.nvim_create_autocmd("BufWritePost", {
  -- Match all `lua` files in `lua/config` except `lazy.lua` which is the
  -- setup file for `lazy.nvim` and should only be reloaded when updated.
  pattern = "**/lua/config/*[^lazy].lua",
  callback = function()
    local filepath = vim.fn.expand("%")
    dofile(filepath)
    vim.notify("Configuration reloaded \n" .. filepath, nil)
  end,
  group = "source config",
  desc = "Reload config on save",
})

```

With this auto command 
I get the "no such mapping error". How to fix it?
Error image:
![image](https://github.com/LazyVim/LazyVim/assets/11295355/5bd118b5-20c6-4ace-a7d1-ae3ca8b97f8a)

**Comment by dpetka2001** [comment, marked as answer]
This is normal. It only happens when you save the `keymaps.lua` file. Since that autocmd will reload the file whenever you save it, it's only natural for that error to occur, because in the `keymaps.lua` file you have already deleted the keybinding and when the file gets reloaded by the autocmd it can't find that same keybinding, since it's already deleted. If you save another file, instead of `keymaps.lua`, you shouldn't see that error. If you don't like it, you could always just remove the autocmd and restart Neovim for the changes in `keymaps.lua` to take effect or manually source the file with `:so %`.

Edit: even if you manually source the file, you'll still get the error since the keybinding won't exist.
